readme rough draft (#11)
1Hive/token-oracle: README.md

# Token Oracle <img align="right" src="https://github.com/1Hive/website/blob/master/website/static/img/bee.png" height="80px" />

> The Token Oracle app is an [ACL Oracle](https://hack.aragon.org/docs/acl_IACLOracle). ACL Oracles are small helper functions that plug in to Aragon's access control list (ACL) to do more sophisticated permission evaluation. In the context of Dandelion Orgs, the Token Oracle is used to check if an address holds Dandelion Org tokens and is thus a member of the organization. This is done by setting a minimum balance in the Token Oracle. Then various Dandelion applications can query the Token Oracle to see if an account submitting the intent to perform an action has the minimum balance of tokens required by the Token Oracle. The Token Oracle will return a boolean which can be used to either approve or reject the intent.

Reviewer: ```suggestion
The Token Oracle app is an [ACL Oracle](https://hack.aragon.org/docs/acl_IACLOracle). ACL Oracles are small helper functions that plug in to Aragon's access control list (ACL) to do more sophisticated permission evaluation. In the context of Dandelion Orgs, the Token Oracle is used to check if an address holds Dandelion Org tokens and is thus a member of the organization. This is done by setting a minimum required balance in the Token Oracle. Then when an account submits the intent to perform an action on an Aragon app function whose ROLE is protected behind this Oracle, the ACL will check with the Oracle if the account has the minimum balance of tokens required. The Token Oracle will return a boolean which can be used to either approve or reject the intent.
```

#### 🐲 Project stage: development

> The Token Oracle app is still in development. If you are interested in contributing please see our open [issues](https://github.com/1hive/time-lock-app/issues).

Reviewer: Would remove this part as it is already fully developed

Reviewer 2: Good call. Do you think we should remove this section entirely, or update to say something like "completed?"

Reviewer: I think it's okay to just remove it.
If we don't explicitly state the project stage it can be assumed as completed. 
Nevertheless if someone wants to check if there are any open issues they can do so in Contributing section

#### 🚨 Security review status: pre-audit

The code in this repo has not been audited.

## How does it work?

> The Token Oracle app is initialized with the address of an ERC-20 token and a minimum balance. It has setters for both of these parameters. Other applications can then query the Token Oracle to determine if an account has at least the minimum balance of the token that the Token Oracle is tracking.

Reviewer: ```suggestion
The Token Oracle app is initialized with the address of an ERC-20 token and a minimum balance. It has setters for both of these parameters. Other applications can then "query" through the ACL, the Token Oracle to determine if an account has at least the minimum balance of the token that the Token Oracle is tracking.
```

## Initialization

The Token Oracle is initialized with `address _token` and `uint256 _minBalance` parameters. The `address _token` parameter is the address of the token that Token Oracle is to track. The `uint256 _minBalance` is the minimum balance of `address _token` that an account needs to hold.

## Roles

The Token Oracle app should implement the following roles:
- **SET_TOKEN_ROLE**: This allows for changing the token address that the Token Oracle tracks.
- **SET_MIN_BALANCE_ROLE**: This allows for changing the threshold at which the Token Oracle returns a true or false boolean.

## Interface

> The Dissent Oracle does not have an interface. It is meant as a back-end helper function for Aragon applications to perform more sophisticated permissions evaluation.

Reviewer: ```suggestion
The Token Oracle does not have an interface. It is meant as a back-end helper function for Aragon applications to perform more sophisticated permissions evaluation.
```

## How to run Token Oracle locally

The Token Oracle works in tandem with other Aragon applications. While we do not explore this functionality as a stand alone demo, the [Dandelion Org template](https://github.com/1Hive/dandelion-org) uses the Token Oracle and it can be run locally.

## Deploying to an Aragon DAO

TBD

## Contributing

We welcome community contributions!

> Please check out our [open Issues](https://github.com/1Hive/time-lock-app/issues) to get started.

Reviewer: ```suggestion
Please check out our [open Issues](https://github.com/1Hive/token-oracle/issues) to get started.
```
